### ISSUE TITLE
Questions about FLOW_ISOLATE ?

### ISSUE THREAD

**Issue by beef9999** [F-Stack/f-stack#759]
@jfb8856606  你好，我使用的是Mellanox网卡，它的驱动可以跟内核共存，所以不需要绑定到uio或者vfio上。

我打算开启DPDK的`FLOW_ISOLATE`模式，将某个端口的TCP报文转发到SR-IOV的VF上，然后f-stack在这个VF上启动。因为我没其他的管理IP或者网卡。

想知道这种情况下，该如何设置`config.ini`？主机IP还需要填吗？能不能给个例子？

--------

Hi, I was using the Mellanox NIC. Its driver can run together with the kernel, so I don't need to bind the NIC to uio/vfio.

Instead, I could enable the `FLOW_ISOLATE` mode, and direct the flow of unique TCP port to a SR-IOV VF. I don't have extra management IP or NICs.

Just want to know how to set `config.ini` in this situation. Is the host IP still required?

**Comment by jfb8856606** [comment]
Hi Bob,

You need enable some argument in `lib/Makefile` and rebuild F-Stack, and then set FDIR policy in `lib/ff_dpdk_if.c`
```
FF_FLOW_ISOLATE=1
FF_FDIR=1
```

发件人: Bob Chen ***@***.***>
发送时间: 2023年4月11日 15:01
收件人: F-Stack/f-stack ***@***.***>
抄送: fengbojiang(姜凤波) ***@***.***>; Mention ***@***.***>
主题: [Internet][F-Stack/f-stack] Questions about FLOW_ISOLATE ? (Issue #759)


@jfb8856606<https://github.com/jfb8856606> 你好，我使用的是Mellanox网卡，它的驱动可以跟内核共存，所以不需要绑定到uio或者vfio上。

我打算开启DPDK的FLOW_ISOLATE模式，将某个端口的TCP报文转发到SR-IOV的VF上。因为我没其他的管理IP或者网卡。

想知道这种情况下，该如何设置config.ini？主机IP还需要填吗？能不能给个例子？

________________________________

Hi, I was using the Mellanox NIC. Its driver can run together with the kernel, so I don't need to bind the NIC to uio/vfio.

Instead, I could enable the FLOW_ISOLATE mode, and direct the flow of unique TCP port to a SR-IOV VF. I don't have extra management IP or NICs.

Just want to know how to set config.ini in this situation. Is the host IP still required?

**Comment by beef9999** [comment]
@jfb8856606  FDIR的本义是Fault detection, isolation, and recovery。我在intel网卡的官方文档没有找到这个词。它把这个称为Flow Director。而Mellanox把它称为Flow Bifurcation。

DPDK上关于FDIR这个词的使用，也只是在中文技术社区的文章里才能搜到，我觉得这个最初的译者可能是把故障恢复和流分类搞混了。

总之，我想问的是f-stack config的IP仍然需要配置成宿主机的IP吗？SR-IOV的VF是否需要配置IP？

------

The original meaning of FDIR is Fault detection, isolation, and recovery. I did not find this word in the official documentation of the intel network card. It calls this Flow Director. And Mellanox calls it Flow Bifurcation.

The use of the word FDIR on DPDK is only mentioned in the Chinese technical community. I think the original translator may have confused fault recovery with flow classification.

In short, what I want to ask is does the IP of f-stack config still need to be configured as the IP of the host? Does the VF of SR-IOV need to configure IP?

**Comment by jfb8856606** [comment]
Hi Bob,

F-Stack config’s addr is needed by FreeBSD network stack, you must config it whether use RSS or Flow Director/Flow Bifurcation.


发件人: Bob Chen ***@***.***>
发送时间: 2023年4月11日 16:28
收件人: F-Stack/f-stack ***@***.***>
抄送: fengbojiang(姜凤波) ***@***.***>; Mention ***@***.***>
主题: [Internet]Re: [F-Stack/f-stack] Questions about FLOW_ISOLATE ? (Issue #759)


@jfb8856606<https://github.com/jfb8856606> FDIR的本义是Fault detection, isolation, and recovery。我在intel网卡的官方文档没有找到这个词。它把这个称为Flow Director。而Mellanox把它称为Flow Bifurcation。

DPDK上关于FDIR这个词的使用，也只是在中文技术社区的文章里才能搜到，我觉得这个最初的译者可能是把故障恢复和流分类搞混了。

总之，我想问的是f-stack config的IP仍然需要配置成宿主机的IP吗？SR-IOV的VF是否需要配置IP？

________________________________

The original meaning of FDIR is Fault detection, isolation, and recovery. I did not find this word in the official documentation of the intel network card. It calls this Flow Director. And Mellanox calls it Flow Bifurcation.

The use of the word FDIR on DPDK is only mentioned in the Chinese technical community. I think the original translator may have confused fault recovery with flow classification.

In short, what I want to ask is does the IP of f-stack config still need to be configured as the IP of the host? Does the VF of SR-IOV need to configure IP?
